go/consensus/tendermint/roothash/roothash.go

```diff
@@ -413,182 +420,181 @@ func (tb *tendermintBackend) onEventDataNewBlock(
-
-				if doBroadcast {
-					notifiers := tb.getRuntimeNotifiers(value.ID)
-					notifiers.eventNotifier.Broadcast(&ev)
-				} else {
-					events = append(events, ev)
-				}
+				ev := api.Event{RuntimeID: value.ID, Height: height, TxHash: txHash, MergeDiscrepancyDetected: &value.Event}
+				events = append(events, ev)
 			case bytes.Equal(key, app.KeyExecutionDiscrepancyDetected):
 				// An execution discrepancy has been detected.
 				var value app.ValueExecutionDiscrepancyDetected
 				if err := cbor.Unmarshal(val, &value); err != nil {
-					tb.logger.Error("failed to unmarshal execution discrepancy from event",
-						"err", err,
-					)
+					errs = multierror.Append(errs, fmt.Errorf("roothash: corrupt ValueExectutionDiscrepancy event: %w", err))
 					continue
 				}
 
-				ev := api.Event{Height: height, TxHash: eh, ExecutionDiscrepancyDetected: &value.Event}
-
-				if doBroadcast {
-					notifiers := tb.getRuntimeNotifiers(value.ID)
-					notifiers.eventNotifier.Broadcast(&ev)
-				} else {
-					events = append(events, ev)
-				}
+				ev := api.Event{RuntimeID: value.ID, Height: height, TxHash: txHash, ExecutionDiscrepancyDetected: &value.Event}
+				events = append(events, ev)
 			case bytes.Equal(key, app.KeyExecutorCommitted):
 				// An executor commit has been processed.
 				var value app.ValueExecutorCommitted
 				if err := cbor.Unmarshal(val, &value); err != nil {
-					tb.logger.Error("failed to unmarshal executor committed event",
-						"err", err,
-					)
+					errs = multierror.Append(errs, fmt.Errorf("roothash: corrupt ValueExecutorCommitted event: %w", err))
 					continue
 				}
 
-				ev := api.Event{Height: height, TxHash: eh, ExecutorCommitted: &value.Event}
-
-				if doBroadcast {
-					notifiers := tb.getRuntimeNotifiers(value.ID)
-					notifiers.eventNotifier.Broadcast(&ev)
-				} else {
-					events = append(events, ev)
-				}
+				ev := api.Event{RuntimeID: value.ID, Height: height, TxHash: txHash, ExecutorCommitted: &value.Event}
+				events = append(events, ev)
 			case bytes.Equal(key, app.KeyMergeCommitted):
 				// A merge commit has been processed.
 				var value app.ValueMergeCommitted
 				if err := cbor.Unmarshal(val, &value); err != nil {
-					tb.logger.Error("failed to unmarshal executor committed event",
-						"err", err,
-					)
+					errs = multierror.Append(errs, fmt.Errorf("roothash: corrupt ValueMergeCommitted event: %w", err))
 					continue
 				}
 
-				ev := api.Event{Height: height, TxHash: eh, MergeCommitted: &value.Event}
-
-				if doBroadcast {
-					notifiers := tb.getRuntimeNotifiers(value.ID)
-					notifiers.eventNotifier.Broadcast(&ev)
-				} else {
-					events = append(events, ev)
-				}
+				ev := api.Event{RuntimeID: value.ID, Height: height, TxHash: txHash, MergeCommitted: &value.Event}
+				events = append(events, ev)
+			case bytes.Equal(key, app.KeyRuntimeID):
+				// Runtime ID attribute.
+				// TODO: Every event should contain this attribute, but not used atm
```

Comment:
See this comments please @kostko  (should we remove the `RuntimeID` field in each of the events, now that we have this attribute? or should we keep it unused for now, maybe just add a check for it to exist for every event?)

Reply:
So the `RuntimeID` field would mostly be used for filtering events in queries (in this case the runtime ID that it contains as value should be text-serialized) not for populating it from anywhere else. So it is fine if it is unused during processing (could add a sanity check to make sure it is there though).